faiss/impl/ResultCollector.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <vector>
+
+#include <faiss/MetricType.h>
+#include <faiss/utils/Heap.h>
+
+/**
+ * ResultCollector is intended to define how to collect search result
+ * For each single search result, collect method will be called.
+ * After every results are collected, post_process method is called at the end.
+ */
+
+namespace faiss {
+
+/** Encapsulates a set of ids to handle. */
+struct ResultCollector {
+    /**
+     * For each result, collect method is called to store result
+     * @param k         number of vectors to search
+     * @param nres      number of results in queue
+     * @param bh_val    search result, distances from query
+     * @param bh_ids    search result, ids of vectors
```

Comment:
What does bh stand for?

Reply:
I don't know. Just kept same variable name as faiss original code.